src/Collections-Strings/String.class.st

```diff
@@ -1874,17 +1874,16 @@ String >> isString [
 
 { #category : #testing }
 String >> isValidGlobalName [
-
+	"Answer whether the string receiver can be used as a global name (e.g., name of a class, name of a trait)"
 	self ifEmpty: [ ^ false ].
 	
 	"reserverd default names"
-	self = 'NameOfSubclass' ifTrue: [ ^ false ].
-	self = 'TNameOfTrait' ifTrue: [ ^ false ].
+	(#('NameOfSubclass' 'MyClass' 'TNameOfTrait') includes: self) ifTrue: [ ^ false ].
```

Comment:
I did not know about this logic. But it's very surprizing. Why we can't have classes with those names? What makes them so special?

Reply:
It would be really nice to explain it in the method comment

Reply:
@bergel also it should be "reserved" - not "reserverd"